custom/lib/Passage.php

```diff
@@ -18,182 +15,293 @@
 
     /**
      * Initialize a new Passage instance.
      *
-     * @param appId  $appId  Passage app id (required)
-     * @param apiKey $apiKey Passage api key (required)
-     *
-     * @throws \InvalidArgumentException
+     * @param string $appId  Passage app id (required)
+     * @param string $apiKey Passage api key (required)
      */
     public function __construct(string $appId, string $apiKey)
     {
         $this->appId = $appId;
-        $this->apiKey = $apiKey;
         $this->clientConfiguration = new Configuration();
         $this->clientConfiguration->setAccessToken($apiKey);
 
-        $this->usersApi = new UsersApi(null, $this->clientConfiguration);
-        $this->userDevicesApi = new UserDevicesApi(null, $this->clientConfiguration);
+        $this->auth = new Auth($appId, $this->clientConfiguration);
+        $this->user = new User($appId, $this->clientConfiguration);
     }
 
     /**
+     * @deprecated 1.0.0 Will not be replaced.
+     *
      * Get AppId key for this Passage instance
      *
      * @return string|null Passage API Key
      */
     public function getAppId(): string
     {
+        trigger_error('Method ' . __METHOD__ . ' is deprecated', E_USER_DEPRECATED);
+
         return $this->appId;
     }
 
     /**
-     * @throws \OpenAPI\Client\ApiException on non-2xx response
-     * @throws \InvalidArgumentException
+     * @deprecated 1.0.0 Will not be replaced.
+     *
      * @return AppInfo|Model401Error|Model404Error|Model500Error
+     * @throws ApiException
      */
     public function getApp(): AppInfo
     {
+        trigger_error('Method ' . __METHOD__ . ' is deprecated', E_USER_DEPRECATED);
+
         $appsApi = new AppsApi();
         return $appsApi->getApp($this->appId)['app'];
     }
 
     /**
-     * Get App for this Passage instance
+     * @deprecated 1.0.0
+     * @see Passage->auth->createMagicLink
      *
+     * Create a Magic Link for your app.
+     *
+     * @param CreateMagicLinkRequest $create_magic_link_request Args for creating a Magic Link
      * @return MagicLink|Model401Error|Model404Error|Model500Error MagicLink object
+     * @throws ApiException
      */
     public function createMagicLink(
         CreateMagicLinkRequest $create_magic_link_request
     ): MagicLink|Model401Error|Model404Error|Model500Error {
+        trigger_error('Method ' . __METHOD__ . ' is deprecated', E_USER_DEPRECATED);
+
         $magicLinksApi = new MagicLinksApi(null, $this->clientConfiguration);
         return $magicLinksApi->createMagicLink($this->appId, $create_magic_link_request)['magic_link'];
     }
 
     /**
-     * Revoke user token for the user
+     * @deprecated 1.0.0
+     * @see Passage->user->revokeRefreshTokens
+     *
+     * Revokes all of a user's Refresh Tokens using their User ID.
      *
+     * @param string $user_id The Passage user ID
      * @return null|Model401Error|Model404Error|Model500Error
+     * @throws ApiException
```

Comment:
What do you think of throwing a `PassageError` instead, & converting from `ApiException` as needed like https://github.com/passageidentity/passage-php/blob/56a6002f3b2d589a29662a51c7a933835ffa8bd2/custom/lib/Auth.php#L125-L129

Reply:
wouldn't that be a breaking change since the original implementation threw `ApiException`?

Reply:
Ah I see, the rest are new classes, but this is annotating an existing class. Will all the public functions throw PassageError in the next major version?

Reply:
Yes all of the new functions are located in the `Auth` and `User` class files (those classes didn't exist before). Each of them should be throwing `PassageError`